### PR TITLE
Upgrade to new node SDK

### PR DESCRIPTION
**Pull request by silesky**
Note: this is a breaking change, as some of the new node types may be more restrictive (for example, we do not allow use IDs or anonymous IDs to be numbers (this is something we're consistent across basically all of our SDKs); there is no messageId option.

should do a major version bump.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -94,14 +93,16 @@
     }
   },
   "scripts": {
-    "build": "shx rm -rf dist && tsc && copyfiles --up 1 \"src/**/*.hbs\" dist",
+    "build": "shx rm -rf dist && npm run build:typedef-test-artifacts && tsc && copyfiles --up 1 \"src/**/*.hbs\" dist && npm run build:telemetry",
+    "build:telemetry": "./bin/run -m=prod",
```
Comment (oscb) [inline package.json:97]: great addition!

**Review by oscb — APPROVED** [review]
The generated types look so much simpler now. 🚢

**Comment by oscb** [comment]
About having this a major. It makes sense to me.
If we want to retain full compat with the old version of node we could set the new one as a different SDK option. Say `analytics-node-next` ? But I know the old library is too barebones and it's on way to EOL. 
Better to explore the idea of multiple version support for future versions of TW.

**Comment by silesky** [comment]
> Better to explore the idea of multiple version support for future versions of TW.
Agreed. The analytics-node to @segment/analytics-node migration is fairly straightforward and it's 95% bw-compat. I think nudging people to upgrade is a big benefit.

**Pull request merged**